src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs

```diff
@@ -426,6 +382,109 @@ public Task DropSubscriptionConnection()
 
             return NoContent();
         }
+
+        [RavenAction("/databases/*/subscriptions/update", "POST", AuthorizationStatus.ValidUser)]
+        public async Task Update()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), null);
+                var options = JsonDeserializationServer.SubscriptionCreationParams(json);
+
+                var id = GetLongQueryString("id", required: false);
+                SubscriptionState state;
+
+                if (id == null)
+                {
+                    state = Database.SubscriptionStorage.GetSubscriptionFromServerStore(options.Name);
```

Comment:
Can this return null?

Reply:
yes this is if the subscription name is nulloremtpty (we are updating by id)
https://github.com/ravendb/ravendb/pull/10479/files#diff-5c857b96c03026c44f6ab3e6c11b0cdaR139

Reply:
no, it will throw SubscriptionDoesNotExistException if ther eis no such subscription 
https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs#L346